etc/restic/_global.env

```diff
@@ -0,0 +1,16 @@
+# B2 credentials.
+# Extracted settings so both systemd timers and user can just source this when want to work on my B2 backup.
+# See https://restic.readthedocs.io/en/latest/030_preparing_a_new_repo.html
+
+# these variables are sourced FIRST, and any values inside of *.env files for
```

Comment:
It would be nicer maybe if instead the `*.env` files sources this global, think "import base class". Then the env files would be self-contained in the sense that you can do

```bash
$ . /etc/restic/example.env
$ restic snapshots
```

instead of having to source 2 files:

```bash
$  . /etc/restic/_global.env
$ . /etc/restic/example.env
$ restic snapshots
```

Being able to do this is very handy when you set up the repo, or want to work with it manually.

Reply:
I thought about that, but what stopped me was that I thought they'd need different syntax; one would have the "export FOO" and the .env would just have the declarations with the export.

I'm not attached to either way, though.

But why would it be "nicer"?

(By the way, they are now self-contained; the minus sign before the path in the .service file means that if it's not present it won't error.)

Reply:
I'm not sure what you mean with different syntax? 

_global.env:
```
B2_ACCOUNT_ID="<b2-account-id>"
B2_ACCOUNT_KEY="<b2-account-key>"
....
```

example.env:
```
source /etc/restic/_global.env

RESTIC_REPOSITORY="b2:<b2-repo-name>"
...
```

and now you only have to source `example.env` to get all settings, and the base vars are in the root `_global.env` so that they can be shared across other *.env files


Its' nicer because you only have to source 1 file to get all vars you need. See example in my comment :)

Reply:
I just re-read it; I see why it's nicer: manual. 

What do you think about the export thing; am I wrong about that?

Reply:
your old files had "export" for the variables; would that be necessary if we sourced it?

Reply:
I'm not sure I understood how you meant. Like this?:

_global.env
```bash
export RESTIC_REPOSITORY
```

example.env:
```bash
RESTIC_REPOSITORY="the value"
```

Reply:
Aha, if `export` is needed. Possibly not, but I was thinking it does not hurt. You never know when you will drop to a subshell, or put a command in a subshell. Then it's nice to retain the variables :)